dask/base.py

```diff
@@ -24,6 +24,9 @@
 __all__ = ("Base", "compute", "normalize_token", "tokenize", "visualize")
 
 
+thread_state = threading.local()
```

Comment:
Why not optionally import `thread_state` from `distributed.utils`? This is only used by distributed, and what the `key` is is defined by it. I'm wary of this tight coupling here.

Reply:
Then we would need to `import distributed` within `dask.base`, which brings with it hefty import time cost.

Generally I've found it annoying to have two collections of global state.  Solidifying within dask/dask makes can make some things simpler.  

I can imagine having the local schedulers also define `thread_state.key` when operating.  It's often useful for complex applications.